classes/core/PKPAppKey.php

```diff
@@ -0,0 +1,185 @@
+    }
+
+    /**
+     * Get the defined cipher
+     */
+    public static function getCipher(): string
+    {
+        return Config::getVar('security', 'cipher', self::$defaultCipher);
+    }
+
+    /**
+     * Has the app key defined in config file
+     */
+    public static function hasKey(): bool
+    {
+        return !empty(Config::getVar('general', 'app_key', ''));
+    }
+
+    /**
+     * Has the app key variable defined in config file
+     */
+    public static function hasKeyVariable(): bool
+    {
+        return Config::hasVar('general', 'app_key');
+    }
+
+    /**
+     * Get the app key defined in config file
+     */
+    public static function getKey(): string
+    {
+        return Config::getVar('general', 'app_key', '');
+    }
+
+    /**
+     * Validate a given cipher
+     */
+    public static function validateCipher(string $cipher): string
+    {
+        $cipher = strtolower($cipher);
+
+        if (!in_array($cipher, array_keys(static::getSupportedCiphers()))) {
+            $ciphers = implode(', ', array_keys(static::getSupportedCiphers()));
+            
+            throw new Exception(
+                sprintf(
+                    'Invalid cipher %s provided, must be among [%s]',
+                    $cipher,
+                    $ciphers
+                )
+            );
+        }
+
+        return $cipher;
+    }
+
+    /**
+     * Validate given or config defined app
+     */
+    public static function validate(string $key = null, string $cipher = null): bool
+    {
+        $config = app('config')->get('app');
+
+        return Encrypter::supported(
+            static::parseKey($key ?? $config['key']),
+            static::validateCipher($cipher ?? $config['cipher'])
+        );
+    }
+
+    /**
+     * Generate a new app key
+     */
+    public static function generate(string $cipher = null): string
+    {
+        $config = app('config')->get('app');
+
+        return 'base64:'.base64_encode(
+            Encrypter::generateKey(static::validateCipher($cipher ?? $config['cipher']))
+        );
+    }
+
+    /**
+     * Write the given app key in the config file
+     * 
+     * @throws \Exception
+     */
+    public static function writeToConfig(string $key): bool
+    {
+        if (!static::validate($key)) {
+            $ciphers = implode(', ', array_keys(static::getSupportedCiphers()));
+            
+            // Error invalid app key
+            throw new Exception(
+                "Unsupported cipher or incorrect key length. Supported ciphers are: {$ciphers}."
+            );
+        }
+
+        $configParser = new ConfigParser;
+        $configParams = [
+            'general' => [
+                'app_key' => $key,
+            ],
+        ];
+
+        if (!static::hasKeyVariable()) {
+            // Error if the config key `app_key` not defined under `general` section
+            throw new Exception('Config variable named `app_key` not defined in the `general` section');
```

Comment:
I think this is redundant with https://github.com/pkp/pkp-lib/pull/9918/files#diff-8ae64e2e6552e29dc508801b292812f26856a2d5e80147614e8b39d48e29d906R36 (I have a suggestion on that line already).

Reply:
I Think we can keep this one as these are 2 different method (one to set/write the key variable and one to set/write the key itself) which can work independent of each other .